src/meta/src/manager/system_param.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use risingwave_pb::meta::SystemParams;
+
+use super::MetaSrvEnv;
+use crate::model::MetadataModel;
+use crate::storage::MetaStore;
+use crate::{MetaError, MetaResult};
+
+pub type SystemParamManagerRef<S> = Arc<SystemParamManager<S>>;
+
+pub struct SystemParamManager<S: MetaStore> {
+    _env: MetaSrvEnv<S>,
+    params: SystemParams,
+}
+
+impl<S: MetaStore> SystemParamManager<S> {
+    /// Return error if `init_params` conflict with persisted system params.
+    pub async fn new(env: MetaSrvEnv<S>, init_params: SystemParams) -> MetaResult<Self> {
```

Comment:
So when meta restarts, it must still bring the exact same params as CLI options as before. Do I understand it correctly? 🤔

Reply:
Yes, but this might not be the most reasonable behavior. System parameters are mutable, so the meta node cannot restart with the same command after some parameters are modified. Perhaps ignore CLI when we find persisted params? cc @fuyufjh

Reply:
The only way I can think of is to add the configuration to etcd manually or by adding a new interface to the meta service and then bring the new code online.

Reply:
> Perhaps ignore CLI when we find persisted params?

+1 and printing a warning about the ignorance should be enough.

Reply:
> > Perhaps ignore CLI when we find persisted params?
> 
> +1 and printing a warning about the ignorance should be enough.

+1

Reply:
> how do we ensure backward compatibility

Newly added params should have default values, so it's not necessary for the users to add them manually. It's essentially the same as process-level config. 

- If a new process config is introduced and it's immutable, then default value will be used if unspecified by the user.
- If it's mutable, the user can restart the process. For system params, SQL commands can be used.